src/screens/movieInfo/MovieInfo.tsx

```diff
@@ -0,0 +1,277 @@
+  revenue: 0,
+};
+
+function MovieInfo() {
+  const { params } =
+    useRoute<
+      RouteProp<MoviesStackParamList, typeof ROUTES.MoviesStack.MovieInfo>
+    >();
+
+  const navigation =
+    useNavigation<
+      NativeStackNavigationProp<
+        MoviesStackParamList,
+        typeof ROUTES.MoviesStack.MovieInfo
+      >
+    >();
+
+  const [movieInfo, setMovieInfo] = useState<FormattedMovieInfo>({} as any);
+  const [movieImages, setMovieImages] = useState<string[]>([]);
+  const [similarMovies, setSimilarMovies] = useState<FormattedMovieData[]>([]);
+
+  const [page, setPage] = useState(1);
+
+  const [isFavourite, setIsFavourite] = useState(false);
+  const [inWatchlist, setInWatchlist] = useState(false);
+
+  const { favouriteMovies, setFavouriteMovies, watchlist, setWatchlist } =
+    useContext(ThemeAndStorageContext);
+
+  const loadSimilarMovies = useCallback(
+    async function loadSimilarMovies(page: number) {
+      console.log('loadSimilarMovies>>>>>>>>');
+      const newSimilarMovies = await getSimiliarMovies(movieInfo.id + '', page);
+      setSimilarMovies([...similarMovies, ...newSimilarMovies]);
+    },
+    [similarMovies, setSimilarMovies],
+  );
+
+  const addToFavouritesWrapper = useCallback(
+    (movieId: string) => {
+      addToFavourites(movieId);
+      setFavouriteMovies([...favouriteMovies, movieId]);
+    },
+    [favouriteMovies, setFavouriteMovies],
+  );
+
+  const removeFromFavouritesWrapper = useCallback(
+    (movieId: string) => {
+      setFavouriteMovies(favouriteMovies.filter(id => id !== movieId));
+
+      removeFromFavourites(movieId);
+    },
+    [favouriteMovies, setFavouriteMovies],
+  );
+
+  const addToWatchlistWrapper = useCallback(
+    (movieId: string) => {
+      addToWatchlist(movieId);
+
+      setWatchlist([...watchlist, movieId]);
+    },
+    [watchlist, setWatchlist],
+  );
+
+  const removeFromWatchlistWrapper = useCallback(
+    (movieId: string) => {
+      setWatchlist(watchlist.filter(id => id !== movieId));
+
+      removeFromWatchlist(movieId);
+    },
+    [watchlist, setWatchlist],
+  );
+
+  useEffect(() => {
+    loadSimilarMovies(page);
+  }, [page]);
+
+  useEffect(() => {
+    console.log('inital loading>>>>>>>>>');
+    (async () => {
+      const movieInfo = await getMovieInfo(params.movieId);
+      setMovieInfo(movieInfo);
+
+      const movieImages = await getMovieImages(params.movieId);
+      setMovieImages(movieImages);
+
+      const similarMovies = await getSimiliarMovies(params.movieId, 1);
+      setSimilarMovies(similarMovies);
+    })();
+  }, []);
+
+  useEffect(() => {
+    const isFavourite = favouriteMovies.some(
+      movieId => movieId === movieInfo.id + '',
+    );
+
+    const inWatchlist = watchlist.some(
+      movieId => movieId === movieInfo.id + '',
+    );
+
+    setInWatchlist(inWatchlist);
+    setIsFavourite(isFavourite);
+  }, [favouriteMovies, watchlist, setInWatchlist, setIsFavourite, movieInfo]);
+
+  return (
```

Comment:
break this in small render components

refer - [COC](https://docs.google.com/document/d/1jVt9hZ0ydQaNt2b5hYdgpuFfd9PSOnZhK9tLSj9xxtI/edit?tab=t.0#heading=h.mvjz6n2qebkp)

Reply:
This still can be broken into smaller components, its fine for now but make sure to follow this in future